### PR TITLE
CIV-13261 Settlement agreement displaying inconsistent dates

### PR DESCRIPTION
**Pull request by kalachandrasekar1**
### JIRA link ###

https://tools.hmcts.net/jira/browse/CIV-13261
https://tools.hmcts.net/jira/browse/CIV-13545

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/test/unit/routes/features/settlementAgreement/respondSettlementAgreementController.test.ts [hunk 1]
```diff
@@ -86,7 +94,37 @@ describe('Respond To Settlement Agreement', () => {
       mockClaim.partialAdmission.paymentIntention.paymentDate = new Date(Date.now());
       mockClaim.partialAdmission.paymentIntention.paymentOption = PaymentOptionType.BY_SET_DATE;
       mockClaim.partialAdmission.howMuchDoYouOwe = new HowMuchDoYouOwe(200, 1000);
-      (getClaimById as jest.Mock).mockResolvedValueOnce(mockClaim);
+      //(getClaimById as jest.Mock).mockResolvedValueOnce(mockClaim);
```
Comment (jeswanth-hmcts) [inline src/test/unit/routes/features/settlementAgreement/respondSettlementAgreementController.test.ts:97]: if comment is not required, please remove the commented code

Comment (kalachandrasekar1) [reply src/test/unit/routes/features/settlementAgreement/respondSettlementAgreementController.test.ts]: removed the commented code

File: src/main/routes/features/settlementAgreement/respondSettlementAgreementController.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ function renderView(form: GenericForm<GenericYesNo>, res: Response, data?: objec
 respondSettlementAgreementController.get(DEFENDANT_SIGN_SETTLEMENT_AGREEMENT, (async (req: Request, res: Response, next: NextFunction) => {
   try {
     const claimId = req.params.id;
-    const claim = await getClaimById(claimId, req, true);
+    const claim = await civilServiceClient.retrieveClaimDetails(claimId, <AppRequest>req);
```
Comment (laarmada) [inline src/main/routes/features/settlementAgreement/respondSettlementAgreementController.ts:32]: with this call we are ignoring the current changes in redis, if exist. Please take care.

File: src/main/services/features/settlementAgreement/respondSettlementAgreementService.ts [hunk 3]
```diff
@@ -2,49 +2,62 @@ import { Request} from 'express';
...
   }
   data = {...data, claimant: claim.getClaimantFullName(), defendant: claim.getDefendantFullName()};
   return data;
 };
 
-function getTextForPayByDate(claim: Claim, lang: string){
+function getTextForPayByDate(claim: Claim, lang: string, isClaimantPlanAccepted: boolean){
+  let paymentDate = claim.getPaymentIntention().paymentDate;
+  if (isClaimantPlanAccepted) {
+    const date = claim.claimantResponse.suggestedPaymentIntention.paymentDate as unknown as PaymentDate;
```
Comment (laarmada) [inline src/main/services/features/settlementAgreement/respondSettlementAgreementService.ts:37]: as unknown... Is no so good to have that in the code, could we discuss on the code review meeting?

**Review by laarmada — CHANGES_REQUESTED** [review]

File: src/main/services/features/claimantResponse/signSettlmentAgreementService.ts [hunk 4]
```diff
@@ -21,6 +21,8 @@ export const getPaymentText = (claim: Claim, req: Request): object => {
       data = getTextForPayByDate(claim, lang, true);
     } else if (claim.getSuggestedPaymentIntentionOptionFromClaimant() == PaymentOptionType.INSTALMENTS) {
       data = getTextForPayByInstallments(claim, lang, true);
+    } else if (claim.getSuggestedPaymentIntentionOptionFromClaimant() == PaymentOptionType.IMMEDIATELY) {
```
Comment (manish14836) [inline src/main/services/features/claimantResponse/signSettlmentAgreementService.ts:24]: Use === instead of ==

**Review by laarmada — DISMISSED** [review]

**Review by manish14836 — DISMISSED** [review]

**Review by paul-pearson — DISMISSED** [review]

File: src/main/services/features/claimantResponse/signSettlmentAgreementService.ts (hunk 4 above)
Comment (kalachandrasekar1) [reply src/main/services/features/claimantResponse/signSettlmentAgreementService.ts]: fixed

**Review by jeswanth-hmcts — DISMISSED** [review]

**Pull request merged**